pkg/pipeline/ChartService.go

```diff
@@ -1228,3 +1263,153 @@ func (impl ChartServiceImpl) JsonSchemaExtractFromFile(chartRefId int) (map[stri
 		return schemajson, nil
 	}
 }
+
+func (impl ChartServiceImpl) CheckChartExists(chartRefId int) error {
+	chartRef, err := impl.chartRefRepository.FindById(chartRefId)
+	if err != nil {
+		impl.logger.Errorw("error in finding ref chart by id", "err", err)
+		return err
+	}
+
+	refChartDir := filepath.Join(string(impl.refChartDir), chartRef.Location)
+	if _, err := os.Stat(refChartDir); os.IsNotExist(err) {
+		_, err = impl.ExtractChartIfMissing(chartRef.ChartData, refChartDir, chartRef.Location)
+		return err
+	}
+	return nil
+}
+
+func (impl *ChartServiceImpl) GetChartLocation(chartName string, chartVersion string) string {
+	var chartLocation string
+
+	chartname := strings.ReplaceAll(chartName, ".", "-")
+	chartname = strings.ReplaceAll(chartname, " ", "_")
+	chartversion := strings.ReplaceAll(chartVersion, ".", "-")
+	if !strings.Contains(chartname, chartversion) {
+		chartLocation = chartname + "_" + chartversion
+	} else {
+		chartLocation = chartname
+	}
+	return chartLocation
+}
+
+func (impl *ChartServiceImpl) ValidateFileUploaded(fileName string) error {
```

Comment:
this is for validating the received file in payload from user.. can we change the uploaded file part to avoid confusion

Reply:
right it can be ValidateUploadedFile